src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -8743,19 +8743,6 @@ MethodTableBuilder::HandleGCForExplicitLayout()
 
     MethodTable *pMT = GetHalfBakedMethodTable();
 
-#ifdef FEATURE_COLLECTIBLE_TYPES
```

Comment:
I want to see the places where these dummy GCDescs are actually needed. I am not able to find them via codereview.

Reply:
@davidwrighton Do you happen to remember what is the path that needed these dummy GCDescs for collectible types? As far as I can tell, they are not needed.

Reply:
The concern is the behavior of go_through_object. go_through_object with the dummy gcdesc works, but if the gc ever calls go_through_object or go_through_object_nostart without checking ContainsPointers is not safe. (go_through_object_cl is safe as it checks ContainsPointers internally).

At one point we did not reliably have a guarantee that go_through_object would not be hit for objects where Collectible was true, but ContainsPointers is false. Looking at the gc codebase, it appears that it may have been changed to make that  invariant true, but I'm not sufficiently familiar with the logic to be sure. For instance, in mark_object_simple1 there is a use of go_through_object not directly protected with a call to ContainsPointers and the logic isn't completely clear about avoiding use of the go_through_object in that case.

Reply:
> For instance, in mark_object_simple1 there is a use of go_through_object not directly protected with a call to ContainsPointers and the logic isn't completely clear about avoiding use of the go_through_object in that case.

Yeah, I agree that the logic is not straightforward for this one - but I believe that it still works fine.

I have changed the assert in `GCDesc::GetCGCDescFromMT` to catch the cases where Collectible was true, but ContainsPointers is false. I hope that we are going to hit intermittently it in the CI if there is any corner case that can sneak through.

If we find that the dummy GCDescs have to be there, we will need to generate them for non-canonical generic types.